app/models/piece.rb

```diff
@@ -7,9 +7,39 @@ def move_piece(new_x, new_y)
     update_attributes(x_cord: new_x, y_cord: new_y)
   end
 
-	def is_obstructed?(x, y)
```

Comment:
wouldn't it be simpler if we keep is_obstructed like so...
`def is_obstructed?(new_x, new_y)`
`    obstructed_horizontally?(new_x, new_y) || obstructed_vertically?(new_x, new_y) || obstructed_diagonally?(new_x, new_y)`
`end`
This way we don't have to use a different obstructed method every time and instead use one for simplification